src/trajectory.jl

```diff
@@ -668,79 +594,63 @@ function generalised_uturn_criterion(rho, p_sharp_minus, p_sharp_plus)
     return (dot(rho, p_sharp_minus) <= 0) || (dot(rho, p_sharp_plus) <= 0)
 end
 
-function isterminated(h::Hamiltonian, t::BinaryTree{T}, args...) where {T<:Union{ClassicNoUTurn, GeneralisedNoUTurn}}
-    return isterminated(h, t)
-end
-
 """
 Recursivly build a tree for a given depth `j`.
 """
-function build_tree(
-    rng::AbstractRNG,
-    nt::NUTS{S,C,I,F},
-    h::Hamiltonian,
-    z::PhasePoint,
-    sampler::AbstractTrajectorySampler,
-    v::Int,
-    j::Int,
-    H0::AbstractFloat,
-) where {I<:AbstractIntegrator,F<:AbstractFloat,S<:AbstractTrajectorySampler,C<:AbstractTerminationCriterion}
+function build_tree(rng, int, tc, h, z, sampler, v, j, H0)
```

Comment:
I think we should keep these type information. It makes the code more readable. and it doesn't hurt to have a few extra lines.

Reply:
I don't mind add them back. I'm not removing them because they take a few lines. I'm following the principle of "annotating types only if they are required for multiple dispatch", which may or may not be appreciated by everyone.

Reply:
Reverted both `build_tree` and `transition`.